crates/core/src/host/wasm_common/module_host_actor.rs

```diff
@@ -466,11 +467,18 @@ impl<T: WasmInstance> WasmModuleInstance<T> {
         let ExecuteResult {
             energy,
             timings,
+            memory_allocation,
             call_result,
         } = result;
 
         self.energy_monitor
             .record_reducer(&energy_fingerprint, energy.used, timings.total_duration);
+        self.energy_monitor
+            .record_reducer_allocated_memory(&energy_fingerprint, memory_allocation);
+        WORKER_METRICS
```

Comment:
Since we are reporting the prometheus metric here, I think we can remove the `record_reducer_allocated_memory` function entirely.

Reply:
Ah, alright.